api/v1beta1/frr_node_state_types.go

```diff
@@ -30,6 +30,7 @@ type FRRNodeStateStatus struct {
 	RunningConfig        string `json:"runningConfig,omitempty"`
 	LastConversionResult string `json:"lastConversionResult,omitempty"`
 	LastReloadResult     string `json:"lastReloadResult,omitempty"`
+	LastHealthcheckReset string `json:"lastHealthcheckReset,omitempty"`
```

Comment:
this is what I initially did, but the problem with a boolean (or a "success" string) is that the controller needs to be aware of what config FRR is running when reconciling, because otherwise the field will be flipped next time the health check runs on an empty config. I thought this is fine to not over-complicate what's going on here and provides enough info ("intersecting" with the other status fields), but it's probably possible to expose the FRRHandler's last applied config and consume it here (by comparing in which "state" the current reconcile is). wdyt?

Reply:
I understand the complication, but we need ot have a way to understand if the current status is "reset because of healthcheck".
Another possible approach is to have the configuration controller to reset the state whenever it applies a new configuration (because in the next status update we would want to see the flag as valid until the next healthcheck).

However, I am not sure I like the number of arrows that we are drawing between all the controllers. 
Basically we have the "status" of the node that is composed of:
- what we get from frr (which kind of belongs to the status controller)
- the conversion failed information, which belongs to the configuration controller
- the healthcheck failure (which belongs the the healthcheck controller and it MAY be contributed by the configuration controller)

I have a feelng that things could be much smoother if we merge the health check and the configuration controller, as they both mess up with the configuration.